client/app/visualizations/map/Editor/StyleSettings.jsx

```diff
@@ -0,0 +1,231 @@
+    url: '//{s}.tiles.wmflabs.org/bw-mapnik/{z}/{x}/{y}.png',
+  },
+  {
+    name: 'OpenStreetMap DE',
+    url: '//{s}.tile.openstreetmap.de/tiles/osmde/{z}/{x}/{y}.png',
+  },
+  {
+    name: 'OpenStreetMap FR',
+    url: '//{s}.tile.openstreetmap.fr/osmfr/{z}/{x}/{y}.png',
+  },
+  {
+    name: 'OpenStreetMap Hot',
+    url: '//{s}.tile.openstreetmap.fr/hot/{z}/{x}/{y}.png',
+  },
+  {
+    name: 'Thunderforest',
+    url: '//{s}.tile.thunderforest.com/cycle/{z}/{x}/{y}.png',
+  },
+  {
+    name: 'Thunderforest Spinal',
+    url: '//{s}.tile.thunderforest.com/spinal-map/{z}/{x}/{y}.png',
+  },
+  {
+    name: 'OpenMapSurfer',
+    url: '//korona.geog.uni-heidelberg.de/tiles/roads/x={x}&y={y}&z={z}',
+  },
+  {
+    name: 'Stamen Toner',
+    url: '//stamen-tiles-{s}.a.ssl.fastly.net/toner/{z}/{x}/{y}.png',
+  },
+  {
+    name: 'Stamen Toner Background',
+    url: '//stamen-tiles-{s}.a.ssl.fastly.net/toner-background/{z}/{x}/{y}.png',
+  },
+  {
+    name: 'Stamen Toner Lite',
+    url: '//stamen-tiles-{s}.a.ssl.fastly.net/toner-lite/{z}/{x}/{y}.png',
+  },
+  {
+    name: 'OpenTopoMap',
+    url: '//{s}.tile.opentopomap.org/{z}/{x}/{y}.png',
+  },
+];
+
+const CustomColorPalette = {
+  White: '#ffffff',
+  ...ColorPalette,
+};
+
+export default function StyleSettings({ options, onOptionsChange }) {
+  const [debouncedOnOptionsChange] = useDebouncedCallback(onOptionsChange, 200);
+
+  return (
+    <React.Fragment>
+      <div className="m-b-15">
+        <label htmlFor="map-editor-tiles">Map Tiles</label>
+        <Select
+          data-test="Map.Editor.Tiles"
+          id="map-editor-tiles"
+          className="w-100"
+          value={options.mapTileUrl}
+          onChange={mapTileUrl => onOptionsChange({ mapTileUrl })}
+        >
+          {map(mapTiles, ({ name, url }) => (
+            <Select.Option key={url} data-test={'Map.Editor.Tiles.' + name}>{name}</Select.Option>
+          ))}
+        </Select>
+      </div>
+
+      <h4 className="m-t-15 m-b-15">Markers</h4>
+
+      <div className="m-b-15">
+        <label htmlFor="map-editor-cluster-markers">
+          <Checkbox
+            id="map-editor-cluster-markers"
+            data-test="Map.Editor.ClusterMarkers"
+            defaultChecked={options.clusterMarkers}
+            onChange={event => onOptionsChange({ clusterMarkers: event.target.checked })}
+          />
+          <span>Cluster Markers</span>
+        </label>
+      </div>
+
+      <div className="m-b-15">
+        <label htmlFor="map-editor-customize-markers">
+          <Checkbox
+            id="map-editor-customize-markers"
+            data-test="Map.Editor.CustomizeMarkers"
+            defaultChecked={options.customizeMarkers}
+            onChange={event => onOptionsChange({ customizeMarkers: event.target.checked })}
+          />
+          <span>Override default style</span>
+        </label>
+      </div>
+
+      {options.customizeMarkers && (
+        <React.Fragment>
```

Comment:
This whole section looks much better 👍 

If it's not too much trouble, it could be great to omit irrelevant colors by shape.
For instance, for "Circle Dot" show only "Border color".

Reply:
@ranbena WDYT - currently custom marker styles are not applied when `Group by` option enabled. Maybe, I should hide entire settings block (including checkbox) when `Group by` enabled?

Reply:
I think it would be best to disable the options and also hide the settings block underneath if selected.

<img width="194" alt="Screen Shot 2019-10-28 at 17 44 56" src="https://user-images.githubusercontent.com/486954/67693699-ccf6e580-f9aa-11e9-8353-7566c98e3857.png">

Reply:
@ranbena (tooltip text might have to be improved though):

![image](https://user-images.githubusercontent.com/12139186/67703521-3d5a3280-f9bc-11e9-8304-c20136815261.png)

Reply:
Upd.: `?` icon and tooltip are shown only when checkbox is disabled